src/Games/NexusMods.Games.StardewValley/Emitters/SMAPIGameVersionDiagnosticEmitter.cs

```diff
@@ -0,0 +1,328 @@
+        var gameVersion = SimplifyVersion(loadout.Installation.Version);
+
+        var smapiMod = loadout.Mods
+            .Where(kv => kv.Value.Enabled)
+            .Select(kv => kv.Value)
+            .FirstOrDefault(mod => mod.Metadata.OfType<SMAPIMarker>().Any());
+
+        if (smapiMod is null)
+        {
+            var smapiModCount = loadout.Mods
+                .Where(kv => kv.Value.Enabled)
+                .Count(kv => kv.Value.Metadata.OfType<SMAPIModMarker>().Any());
+
+            // NOTE(erri120): The MissingSMAPIEmitter will warn the user if SMAPI is required.
+            // This emitter will suggest SMAPI if there are no mods yet.
+            if (smapiModCount != 0) yield break;
+
+            var suggestion = SuggestSMAPI(gameToSMAPIMappings, gameVersion);
+            if (suggestion is not null) yield return suggestion;
+            yield break;
+        }
+
+        var smapiMarker = smapiMod.Metadata.OfType<SMAPIMarker>().First();
+
+        // var smapiVersion = SimplifyVersion(new Version("3.18.6"));
+        var smapiVersion = SimplifyVersion(smapiMarker.Version!);
+
+        if (!smapiToGameMappings.TryGetValue(smapiVersion, out var supportedGameVersions))
+        {
+            // ReSharper disable once InconsistentLogPropertyNaming
+            _logger.LogWarning("Found no game version information for SMAPI version {SMAPIVersion}", smapiVersion);
+            yield break;
+        }
+
+        var diagnostic1 = GameVersionOlderThanMinimumGameVersion(
+            gameToSMAPIMappings,
+            loadout, smapiMod,
+            gameVersion, smapiVersion, supportedGameVersions
+        );
+
+        var diagnostic2 = GameVersionNewerThanMaximumGameVersion(
+            gameToSMAPIMappings,
+            loadout, smapiMod,
+            gameVersion, smapiVersion, supportedGameVersions
+        );
+
+        if (diagnostic1 is not null) yield return diagnostic1;
+        if (diagnostic2 is not null) yield return diagnostic2;
+    }
+
+    private Diagnostic? SuggestSMAPI(GameToSMAPIMapping gameToSMAPIMappings, Version gameVersion)
+    {
+        var supportedSMAPIVersion = GetSuggestedSMAPIVersion(gameToSMAPIMappings, gameVersion);
+        if (supportedSMAPIVersion is null) return null;
+
+        return Diagnostics.CreateSuggestSMAPIVersion(
+            LatestSMAPIVersion: supportedSMAPIVersion.ToString(),
+            CurrentGameVersion: gameVersion.ToString(),
+            SMAPINexusModsLink: NexusModsSMAPILink
+        );
+    }
+
+    private Diagnostic? GameVersionNewerThanMaximumGameVersion(
+        GameToSMAPIMapping gameToSMAPIMappings,
+        Loadout loadout,
+        Mod smapiMod,
+        Version gameVersion,
+        Version smapiVersion,
+        GameVersions supportedGameVersions)
+    {
+        var (_, maximumGameVersion) = supportedGameVersions;
+        if (maximumGameVersion is null) return null;
+
+        if (maximumGameVersion > gameVersion) return null;
```

Comment:
![image](https://github.com/Nexus-Mods/NexusMods.App/assets/26797547/933b7ab8-3e13-48da-9277-38f0a849fcb5)
This check fails because it is doing `1.5.6 > 1.5.6.22018` the revision number should be stripped when doing the comparison.
Also this check should be inclusive:
```csharp
if (maximumGameVersion >= gameVersion) return null;
```